interface/modules/custom_modules/oe-module-ehi-exporter/src/Bootstrap.php

```diff
@@ -0,0 +1,211 @@
+
+    /**
+     * @var GlobalConfig Holds our module global configuration values that can be used throughout the module.
+     */
+    private $globalsConfig;
+
+    /**
+     * @var string The folder name of the module.  Set dynamically from searching the filesystem.
+     */
+    private $moduleDirectoryName;
+
+    /**
+     * @var \Twig\Environment The twig rendering environment
+     */
+    private $twig;
+
+    /**
+     * @var SystemLogger
+     */
+    private $logger;
+
+    private static self $instance;
+
+    public function __construct(EventDispatcherInterface $eventDispatcher, ?Kernel $kernel = null)
+    {
+        if (empty($kernel)) {
+            $kernel = new Kernel();
+        }
+
+        // NOTE: eventually you will be able to pull the twig container directly from the kernel instead of instantiating
+        // it here.
+        $twig = new TwigContainer($this->getTemplatePath(), $kernel);
+        $twigEnv = $twig->getTwig();
+        $this->twig = $twigEnv;
+
+        $this->moduleDirectoryName = basename(dirname(__DIR__));
+        $this->eventDispatcher = $eventDispatcher;
+
+        // we inject our globals value.
+        $this->globalsConfig = new GlobalConfig($GLOBALS);
+        $this->logger = new SystemLogger();
+    }
+
+    public static function instantiate(EventDispatcher $eventDispatcher, Kernel $kernel): self
+    {
+        if (!isset(self::$instance)) {
+            self::$instance = new Bootstrap($eventDispatcher, $kernel);
+            self::$instance->subscribeToEvents();
+        }
+        return self::$instance;
+    }
+
+    public function getLogger()
+    {
+        return new SystemLogger();
+    }
+
+    public function getExporter()
+    {
+        $xmlConfigPath = $GLOBALS['webserver_root'] . DIRECTORY_SEPARATOR . 'Documentation' . DIRECTORY_SEPARATOR . 'EHI_Export'
+            . DIRECTORY_SEPARATOR . 'docs' . DIRECTORY_SEPARATOR . 'openemr.openemr.xml';
+        return new EhiExporter(
+            $GLOBALS['webserver_root'] . $this->getPublicPath(),
+            $this->getPublicPath(),
+            $xmlConfigPath,
+            $this->getTwig()
+        );
+    }
+
+    public function getTwig()
+    {
+        $container = new TwigContainer($this->getTemplatePath(), $GLOBALS['kernel']);
+        return $container->getTwig();
+    }
+
+    public function subscribeToEvents()
+    {
+//        $this->addGlobalSettings();
+        $this->eventDispatcher->addListener(MenuEvent::MENU_UPDATE, [$this, 'addCustomModuleMenuItem']);
+    }
+
+    public function addCustomModuleMenuItem(MenuEvent $event)
+    {
+        $menu = $event->getMenu();
+
+        $menuItem = new \stdClass();
+        $menuItem->requirement = 0;
+        $menuItem->target = 'msc';
+        $menuItem->menu_id = 'ehiExporter0';
+        $menuItem->label = xlt("Electronic Health Information Export");
+        $menuItem->url = $GLOBALS['webroot'] . "/interface/modules/custom_modules/" . self::MODULE_NAME . "/public/";
```

Comment:
No need to add root because the url is added to root in menu parser so
` $menuItem->url = "/interface/modules/custom_modules/" . self::MODULE_NAME . "/public/";`

Reply:
K, removed in the updated PR.